umodbus/asynchronous/tcp.py

```diff
@@ -0,0 +1,344 @@
+#!/usr/bin/env python3
+#
+# Copyright (c) 2019, Pycom Limited.
+#
+# This software is licensed under the GNU GPL version 3 or any
+# later version, with permitted additional terms. For more information
+# see the Pycom Licence v1.0 document supplied with this file, or
+# available at https://www.pycom.io/opensource/licensing
+#
+
+# system packages
+try:
+    import uasyncio as asyncio
+except ImportError:
+    import asyncio
+
+from ..sys_imports import struct, List, Optional, Tuple
+from ..sys_imports import Callable, Coroutine, Any, Dict
+
+# custom packages
+from .modbus import AsyncModbus
+from .common import AsyncRequest, CommonAsyncModbusFunctions
+from .. import functions, const as Const
+from ..common import ModbusException
+from ..tcp import CommonTCPFunctions, TCPServer
+
+
+class AsyncModbusTCP(AsyncModbus):
+    """
+    Asynchronous equivalent of ModbusTCP class.
+
+    @see ModbusTCP
+    """
+    def __init__(self, addr_list: Optional[List[int]] = None):
+        super().__init__(
+            # set itf to AsyncTCPServer object
+            AsyncTCPServer(),
+            addr_list
+        )
+
+    async def bind(self,
+                   local_ip: str,
+                   local_port: int = 502,
+                   max_connections: int = 10) -> None:
+        """@see ModbusTCP.bind"""
+
+        await self._itf.bind(local_ip, local_port, max_connections)
+
+    def get_bound_status(self) -> bool:
+        """@see ModbusTCP.get_bound_status"""
+
+        return self._itf.is_bound
+
+    async def serve_forever(self) -> None:
+        """
+        Starts serving the asynchronous server on the specified host and port
+        specified in the constructor.
+        """
+
+        await self._itf.serve_forever()
+
+    def server_close(self) -> None:
+        """Stops the server."""
+
+        self._itf.server_close()
+
+
+class AsyncTCP(CommonTCPFunctions, CommonAsyncModbusFunctions):
+    """
+    Asynchronous equivalent of TCP class.
+
+    @see TCP
+    """
+    def __init__(self,
+                 slave_ip: str,
+                 slave_port: int = 502,
+                 timeout: float = 5.0):
+        """
+        Initializes an asynchronous TCP client. 
```

Comment:
```suggestion
        Initializes an asynchronous TCP client.
```
`flake8` reported a trailing whitespace